gst/tensor_decoder/tensordec.c

```diff
@@ -512,10 +521,16 @@ gst_tensordec_set_property (GObject * object, guint prop_id,
       self->silent = g_value_get_boolean (value);
       break;
     case PROP_MODE:
-      self->mode = g_value_dup_string (value);
+      if (g_strcmp0 (g_value_dup_string (value), "image_labeling") == 0) {
+        self->mode = IMAGE_LABELING;
+      } else if (g_strcmp0 (g_value_dup_string (value), "boundig_boxes") == 0) {
+        self->mode = BOUNDING_BOXES;
+      } else {
+        self->mode = DIRECT_VIDEO;
+      }
```

Comment:
Reuse the internal API, ```find_key_strv()``` in tensor_common.h.

With
```
static const gchar *mode_names[] = {
  "direct_video",
  "image_labeling",
  "bounding_boxes",
  NULL
};
```
, you can have ```self->mode = find_key_strv (mode_names, key_name);```

Besides, ```g_value_dup_string()``` does "malloc". Make sure that the returned value is deallocated.

Reply:
I updated code with your recommands
including g_value_dup_string() deallocation issue

Reply:
You don't need these ```if-else``` statements.
Remove Line 527-533, replace it with ```self->mode = find_key_strv (mode_names, temp_string);```